client/components/changed-badge.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+const ChangedBadge = ({ fields = [], latest = [], granted = [] }) => {
+  if (fields.some(field => latest.some(change => change.includes(field)))) {
```

Comment:
i didn't actually look at this code in great detail, but I think you want `_.intersection(fields, changes).length`

Reply:
I'm totally relaxed about whether or not we leave the "double-some" or use `intersection` (it's broadly the same code under the hood either way), but I have strong feelings about putting this much code _inside_ the condition of an `if`.

It forces the user to understand the code in order to parse the condition, where a descriptively named intermediate value can allow a reader to understand the condition without having to mentally reverse-engineer the code. 

```js
const hasChange = /* whatever implementation you choose */;
if (hasChange) {
  /// stuff
}
```

Reply:
#bemorephil